### ISSUE TITLE
Use [scripts.test]` for `anchor test`?

### ISSUE THREAD

**Issue by fanatid** [project-serum/anchor#555]
Source: https://github.com/project-serum/anchor/pull/550#issuecomment-888098045

Proposed breaking change:
- Remove `mocha`/`ts-mocha` selection from `anchor-cli`.
- Use `[scripts.test]` as command for `anchor test`.
- Add `mocha`/`ts-mocha` command to `[scripts.test]` in templates.

**Comment by armaniferrante** [comment]
What if we re-use the existing `[scripts]` section? There are other commands like this where we probably want to be able to add hooks, like `anchor build` and so having one table could be more convenient instead of one per override.

```toml
[scripts]
test = "<test-cmd-here>"
```

**Comment by fanatid** [comment]
Sorry, that did not make my initial message clear! `[scripts.test]` is confusing. I meant exactly what you pasted:
```
[scripts]
test = "<test-cmd-here>"
```

Should we remove `mocha`/`ts-mocha` selection? (i.e. always use value from `[scripts]` section)

**Comment by armaniferrante** [comment]
> Sorry, that did not make my initial message clear! `[scripts.test]` is confusing. I meant exactly what you pasted:
> 
> ```
> [scripts]
> test = "<test-cmd-here>"
> ```
> 
> Should we remove `mocha`/`ts-mocha` selection? (i.e. always use value from `[scripts]` section)

My weak preference is to use it as a default. If [scripts] + test is provided, then to use that as an override. Any reasons to not do this?

**Comment by fanatid** [comment]
I see only one reason -- it's clear for the developer what command is executed. There also will no surprise that something is broken if `[scripts] + test` will be added. But this will introduce breaking change :disappointed:

**Comment by armaniferrante** [comment]
> I see only one reason -- it's clear for the developer what command is executed. There also will no surprise that something is broken if `[scripts] + test` will be added. But this will introduce breaking change

I do like the explicit nature of this change. If made, we should bundle it with other breaking changes. :)

**Comment by fanatid** [comment]
Resolved with #550
